libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -20,6 +23,35 @@ void VotePacketHandler::validatePacketRlpFormat([[maybe_unused]] const PacketDat
   }
 }
 
+bool VotePacketHandler::isVoteRoundStepRelevant(const std::shared_ptr<Vote> &vote) const {
+  auto checking_round = pbft_mgr_->getPbftRound();
+  // If period is not the same we assuming current round is equal to 0
+  // So we won't accept votes for future period with round bigger than kVoteAcceptingSteps
+  if (pbft_chain_->getPbftChainSize() != vote->getPeriod()) {
+    checking_round = 0;
+  }
+  const auto vote_hash = vote->getHash();
+  if (vote->getRound() > checking_round + kVoteAcceptingRounds) {
+    LOG(log_nf_) << "Skip vote " << vote_hash.abridged() << ". Round too big " << pbft_mgr_->getPbftRound() << " "
+                 << vote->getRound();
+    return false;
+  }
+
+  auto checking_step = pbft_mgr_->getPbftStep();
+  // If round is not the same we assuming current step is equal to 0
+  // So we won't accept votes for future rounds with step bigger than kVoteAcceptingSteps
+  if (pbft_mgr_->getPbftRound() != vote->getRound()) {
```

Comment:
dont call this obver and over, you should get it just once as I mentioned in one of the comments